WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StatsTimeRangeBarViewModel.swift

```diff
@@ -80,16 +81,19 @@ struct StatsTimeRangeBarViewModel: Equatable {
         timeRangeText = timeRange.timeRangeText(startDate: startDate,
                                                 endDate: endDate,
                                                 timezone: timezone)
+        selectedDateText = nil
     }
 
     init(startDate: Date,
          endDate: Date,
          selectedDate: Date,
          timeRange: StatsTimeRangeV4,
          timezone: TimeZone) {
-        // Disable editing time range when selecting a specific date on the graph
-        isTimeRangeEditable = false
+        isTimeRangeEditable = timeRange.isCustomTimeRange
         timeRangeText = timeRange.timeRangeText(startDate: startDate,
+                                                endDate: endDate,
+                                                timezone: timezone)
+        selectedDateText = timeRange.timeRangeText(startDate: startDate,
                                                 endDate: endDate,
                                                 selectedDate: selectedDate,
                                                 timezone: timezone)
```

Comment:
I'd suggest keeping the old logic for the other tabs and add the separated selected date label only for the custom range tab. This is what I'd do:

```suggestion
        if timeRange.isCustomTimeRange {
            timeRangeText = timeRange.timeRangeText(startDate: startDate,
                                                    endDate: endDate,
                                                    timezone: timezone)
            selectedDateText = timeRange.timeRangeText(startDate: startDate,
                                                       endDate: endDate,
                                                       selectedDate: selectedDate,
                                                       timezone: timezone)
        } else {
            /// Shows the selected date in place of the time range label for non-custom range tabs.
            timeRangeText = timeRange.timeRangeText(startDate: startDate,
                                                    endDate: endDate,
                                                    selectedDate: selectedDate,
                                                    timezone: timezone)
            selectedDateText = nil
        }
```

Reply:
Yup, this makes sense! Thanks!